spec/helpers/govuk_skip_link_helper_spec.rb

```diff
@@ -26,8 +27,40 @@
     end
 
     context 'when href provided' do
-      subject { govuk_skip_link href: custom_href }
+      subject { govuk_skip_link(href: custom_href) }
       it { is_expected.to have_tag('a', with: { href: custom_href, class: 'govuk-skip-link' }, text: default_link_text) }
     end
+
+    describe 'extra classes' do
+      context 'when supplied with extra classes as a string' do
+        let(:custom_classes) { 'pink' }
+        subject { govuk_skip_link(classes: custom_classes) }
+        it { is_expected.to have_tag('a', with: { href: default_href, class: %w(govuk-skip-link).append(custom_classes) }, text: default_link_text) }
+      end
+
+      context 'when supplied with extra classes as an array' do
+        let(:custom_classes) { %w(yellow spots) }
+        subject { govuk_skip_link(classes: custom_classes) }
+        it { is_expected.to have_tag('a', with: { href: default_href, class: %w(govuk-skip-link).append(custom_classes) }, text: default_link_text) }
+      end
+    end
+
+    describe 'custom html attributes' do
+      let(:custom_attributes) { { lang: "en-GB", data: { awesome: "yes" } } }
+      subject { govuk_skip_link(**custom_attributes) }
+      it { is_expected.to have_tag('a', with: { href: default_href, lang: "en-GB", "data-awesome" => "yes" }, text: default_link_text) }
+    end
+
+    describe 'custom HTML' do
+      let(:custom_text) { "skip to the content" }
+      let(:custom_class) { "yellow" }
+      subject { govuk_skip_link { tag.span(custom_text, class: custom_class) } }
+
+      it "inserts the custom HTML into the anchor element" do
+        expect(subject).to have_tag('a', with: { href: default_href }) do
```

Comment:
Does `is_expected.to` style work with blocks?

Reply:
Sorry missed this question - I don't believe it's possible.